flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -551,6 +551,17 @@ public Table from(String path) {
                                                 "Table %s was not found.", unresolvedIdentifier)));
     }
 
+    @Override
+    public Table from(TableDescriptor descriptor) {
+        final String path = TableDescriptorUtil.getUniqueAnonymousPath();
+
+        final ObjectIdentifier identifier =
+                catalogManager.qualifyIdentifier(UnresolvedIdentifier.of(path));
+        createTemporaryTable(identifier.toObjectPath().getFullName(), descriptor);
```

Comment:
The `toObjectPath().getFullName()` and then reparsing seems error-prone to me. How about introducing a `createTemporaryTableInternal` that takes an `UnresolvedIdentifier` for both `from` and `createTemporaryTable`.